packages/db/postgres/tables.sql

```diff
@@ -0,0 +1,159 @@
+  -- Identifier for the service that is pinning this pin.
+  pin_location_id BIGINT                                                        NOT NULL REFERENCES pin_location (id),
+  inserted_at     TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  updated_at      TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  UNIQUE (content_cid, pin_location_id)
+);
+
+CREATE TYPE upload_type AS ENUM
+(
+  -- A CAR file upload.
+  'Car',
+  -- Files uploaded and converted into a CAR file.
+  'Upload',
+  -- A raw blob upload in the request body.
+  'Blob',
+  -- A multi file upload using a multipart request.
+  'Multipart'
+);
+
+-- An upload created by a user.
+CREATE TABLE IF NOT EXISTS upload
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- User that uploaded this content.
+  user_id         BIGINT                                                        NOT NULL REFERENCES public.user (id),
+  -- User authentication token that was used to upload this content.
+  -- Note: nullable, because the user may have used a Magic.link token.
+  auth_key_id     BIGINT REFERENCES auth_key (id),
+  -- The root of the uploaded content.
+  content_cid     TEXT                                                          NOT NULL REFERENCES content (cid),
+  -- CID for the content as provided by the user.
+  -- TODO: do we need this? Should we fail if they do not match?
+  source_cid      TEXT                                                          NOT NULL,
+  -- Type of received upload data.
+  type            upload_type                                                   NOT NULL,
+  -- User provided name for this upload.
+  name            TEXT,
+  inserted_at     TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  updated_at      TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  deleted_at      TIMESTAMP WITH TIME ZONE,
+  UNIQUE (user_id, content_cid)
+);
+
+-- Details of the backups created for an upload.
+CREATE TABLE IF NOT EXISTS backup
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- Upload that resulted in this backup.
+  upload_id       BIGINT                                                        NOT NULL REFERENCES upload (id),
+  -- Backup url location.
+  url             TEXT                                                          NOT NULL UNIQUE,
+  inserted_at     TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  deleted_at      TIMESTAMP WITH TIME ZONE
+);
+
+-- A request to keep a Pin in sync with the nodes that are pinning it.
+CREATE TABLE IF NOT EXISTS pin_request
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- Identifier for the pin to keep in sync.
+  pin_id          BIGINT                                                        NOT NULL REFERENCES pin (id),
+  inserted_at     TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL
```

Comment:
we dont delete pin_requests?

Reply:
Good question!! I think we should delete, but apparently we do not have it in our fauna schema: https://github.com/web3-storage/web3.storage/blob/main/packages/db/fauna/schema.graphql#L176 😱 

Ideas why?

Reply:
Added as I think this make sense!